src/GafferScene/EvaluateLightLinks.cpp

```diff
@@ -142,19 +146,21 @@ IECore::ConstCompoundObjectPtr EvaluateLightLinks::computeAttributes( const Scen
 	{
 		result->members()[ m_lightLinkAttrName ] = expressionToLightNames( illuminationExpressionData->readable(), inPlug(), lightsSet );
 	}
-	else
+	else if( path.size() == 1 )
 	{
-		// \todo: this transformation should probably get cached?
 		ConstPathMatcherDataPtr defaultLightsSetData = inPlug()->set( m_defaultLightSetName );
 		const PathMatcher &defaultLightsSet = defaultLightsSetData->readable();
 
-		IECore::StringVectorDataPtr defaultLightNames = new IECore::StringVectorData();
-		std::vector<std::string> &defaultLightNamesWritable = defaultLightNames->writable();
+		if( defaultLightsSet != lightsSet )
+		{
+			IECore::StringVectorDataPtr defaultLightNames = new IECore::StringVectorData();
```

Comment:
For what it's worth, I've adopted a convention of `<Type>DataPtr fooData; <type> &foo;`, just because it gives a shorter name for the variable that typically gets used more.

Reply:
```
template<typename T>
class DataReadWriter
{
public:

	using Ptr = typename T::Ptr;
	using Value = typename T::ValueType;

	DataReadWriter( IECore::DataPtr ptr ) : m_ptr( IECore::runTimeCast<T>(ptr) ), rw( m_ptr->writable() ) {}

private:
	Ptr m_ptr;

public:
	Value &rw;
};

template<typename T>
class DataReader
{
public:
	using ConstPtr = typename T::ConstPtr;
	using Value = typename T::ValueType;

	DataReader( IECore::ConstDataPtr ptr ) : m_ptr( IECore::runTimeCast<const T>( ptr ) ), r( m_ptr->readable() ) {}

private:
	ConstPtr m_ptr;
public:
	const Value &r;
};
```

This suggestion doesn't address variable name length but I was just musing about how to reduce the need to have two locals : fooData & foo.